src/qt/transactiontablemodel.cpp

```diff
@@ -193,8 +192,8 @@ class TransactionTablePriv
             // simply re-use the cached status.
             interfaces::WalletTxStatus wtx;
             int numBlocks;
-            if (wallet.tryGetTxStatus(rec->hash, wtx, numBlocks) && rec->statusUpdateNeeded(numBlocks)) {
-                rec->updateStatus(wtx, numBlocks);
+            if (wallet_model->wallet().tryGetTxStatus(rec->hash, wtx, numBlocks) && rec->statusUpdateNeeded(numBlocks)) {
+                rec->updateStatus(wtx, numBlocks, QDateTime::fromTime_t(wallet_model->node().getLastBlockTime()));
```

Comment:
Should avoid evaluating `QDateTime::fromTime_t(wallet_model->node().getLastBlockTime())` for each record mostly because of excessive `cs_main` locks.